pkg/ovs/ovsconfig/ovs_schema.go

```diff
@@ -14,27 +14,184 @@
+}
+
+type (
+	BridgeFailMode  = string
+	BridgeProtocols = string
+)
+
+var (
+	BridgeFailModeSecure      BridgeFailMode  = "secure"
+	BridgeFailModeStandalone  BridgeFailMode  = "standalone"
+	BridgeProtocolsOpenflow10 BridgeProtocols = "OpenFlow10"
+	BridgeProtocolsOpenflow11 BridgeProtocols = "OpenFlow11"
+	BridgeProtocolsOpenflow12 BridgeProtocols = "OpenFlow12"
+	BridgeProtocolsOpenflow13 BridgeProtocols = "OpenFlow13"
+	BridgeProtocolsOpenflow14 BridgeProtocols = "OpenFlow14"
+	BridgeProtocolsOpenflow15 BridgeProtocols = "OpenFlow15"
+)
+
+// Bridge defines an object in Bridge table
 type Bridge struct {
-	Name         string        `json:"name"`
-	Protocols    []interface{} `json:"protocols,omitempty"`
-	DatapathType string        `json:"datapath_type,omitempty"`
+	UUID                string            `ovsdb:"_uuid"`
+	AutoAttach          *string           `ovsdb:"auto_attach"`
+	Controller          []string          `ovsdb:"controller"`
+	DatapathID          *string           `ovsdb:"datapath_id"`
+	DatapathType        string            `ovsdb:"datapath_type"`
+	DatapathVersion     string            `ovsdb:"datapath_version"`
+	ExternalIDs         map[string]string `ovsdb:"external_ids"`
+	FailMode            *BridgeFailMode   `ovsdb:"fail_mode"`
+	FloodVLANs          []int             `ovsdb:"flood_vlans"`
+	FlowTables          map[int]string    `ovsdb:"flow_tables"`
+	IPFIX               *string           `ovsdb:"ipfix"`
+	McastSnoopingEnable bool              `ovsdb:"mcast_snooping_enable"`
+	Mirrors             []string          `ovsdb:"mirrors"`
+	Name                string            `ovsdb:"name"`
+	Netflow             *string           `ovsdb:"netflow"`
+	OtherConfig         map[string]string `ovsdb:"other_config"`
+	Ports               []string          `ovsdb:"ports"`
+	Protocols           []BridgeProtocols `ovsdb:"protocols"`
+	RSTPEnable          bool              `ovsdb:"rstp_enable"`
+	RSTPStatus          map[string]string `ovsdb:"rstp_status"`
+	Sflow               *string           `ovsdb:"sflow"`
+	Status              map[string]string `ovsdb:"status"`
+	STPEnable           bool              `ovsdb:"stp_enable"`
 }
+type (
+	PortBondMode = string
+	PortLACP     = string
+	PortVLANMode = string
+)
+
+var (
+	PortBondModeActiveBackup   PortBondMode = "active-backup"
+	PortBondModeBalanceSLB     PortBondMode = "balance-slb"
+	PortBondModeBalanceTCP     PortBondMode = "balance-tcp"
+	PortLACPActive             PortLACP     = "active"
+	PortLACPOff                PortLACP     = "off"
+	PortLACPPassive            PortLACP     = "passive"
+	PortVLANModeAccess         PortVLANMode = "access"
+	PortVLANModeDot1qTunnel    PortVLANMode = "dot1q-tunnel"
+	PortVLANModeNativeTagged   PortVLANMode = "native-tagged"
+	PortVLANModeNativeUntagged PortVLANMode = "native-untagged"
+	PortVLANModeTrunk          PortVLANMode = "trunk"
+)
 
+// Port defines an object in Port table
 type Port struct {
```

Comment:
Can we keep the current model and only include the fields that we actually need?
Some fields may be required but I am pretty sure not all fields are required.
When I was working https://github.com/antrea-io/ofnet/pull/76, I only had a subset of fields for the `bridge` table.

Reply:
yes, I just used the schema generated by model-gen to experiment with the flow. Will also configure monitor options for selected fields only rather than the complete table.